io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingTest.java

```diff
@@ -320,6 +322,28 @@ public void testNestedJoinHandler() {
 		assertEquals(SOURCE_DATASET_ID, nestedMapping.getSourceExpression().getOCQL());
 		assertNull(nestedMapping.getSourceExpression().getLinkElement());
 		assertNull(nestedMapping.getSourceExpression().getLinkField());
+
+		// check namespace correct prefix output
+		List<Namespace> namespaces = mainMapping.getNamespaces().getNamespace();
+		checkExpectedNamespacesPrefixes(namespaces);
+	}
+
+	private void checkExpectedNamespacesPrefixes(List<Namespace> namespaces) {
```

Comment:
Please add java-doc, this shoudl a generic helper method receives a map containing the expected namespaces and a map containing the current namespaces.